packages/ra-core/src/ui/detail/Show.js

```diff
@@ -134,52 +99,32 @@ export class Show extends Component {
         const defaultTitle = translate('ra.page.show', {
             name: `${resourceName}`,
             id,
-            data,
+            record,
+        });
+        return children({
+            actionProps: {
```

Comment:
will recreate a new object at each render, and therefore miss the identity check. You should inline them all.

Reply:
This shouldn't be done by CoreShow at all actually. Moving it to the mui Show